### PR TITLE
Split Network.VB into 7 Files from PR #11863

### PR DESCRIPTION
**Pull request by paul1956**
## Proposed changes
- Split Network.Vb in 7 files to prepare for replacement of WebClient for download

## Customer Impact
- Developer only improvement 

## Regression? 
- No

## Risk
- Minimal all tests from #11863 pass

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12226)

### CODE REVIEW

**Comment by paul1956** [comment]
Reviewer note: coverage is identical to current unmodified code. The tests are in a separate PR #12221 .

**Comment by paul1956** [comment]
@KlausLoeffelmann there are no code changes in this PR, it just splits 1 large file into logical parts. Changes happen in follow on PR's. Having someone else do review might give you more time to work on other projects.

**Comment by paul1956** [comment]
@lonitra network.vb being completely rewritten in PR #11863 to replace WebClient which is obsolete. This allows the changes to be more isolated and the reviews simpler.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkPing.vb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+
+Imports NetInfoAlias = System.Net.NetworkInformation
+Imports VbUtils = Microsoft.VisualBasic.CompilerServices.ExceptionUtils
+
+Namespace Microsoft.VisualBasic.Devices
+
+    Partial Public Class Network
```
[the remark this answers is not in the document]
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkPing.vb:9]: Thank you for the explanation @paul1956 ! In that case could we maybe name these files Network.Ping.vb to be clear that this is partial class for network and not a new NetworkPing class?

[the remark this answers is not in the document]
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkPing.vb:9]: @lonitra great suggestion. I will update today.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkAvailability.vb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+Imports System.Threading
+
+Imports NetInfoAlias = System.Net.NetworkInformation
+
+Namespace Microsoft.VisualBasic.Devices
+
+    <EditorBrowsable(EditorBrowsableState.Advanced)>
+    Public Delegate Sub NetworkAvailableEventHandler(sender As Object, e As NetworkAvailableEventArgs)
+
+    Partial Public Class Network
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkAvailability.vb:15]: Consider renaming this file as well to Network.Availability.vb

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.UploadFile.vb [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+' The .NET Foundation licenses this file to you under the MIT license.
+
+Imports System.Net
+Imports Microsoft.VisualBasic.CompilerServices
+Imports Microsoft.VisualBasic.FileIO
+Imports Microsoft.VisualBasic.MyServices.Internal
+
+Namespace Microsoft.VisualBasic.Devices
+
+    Public Class Network
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.UploadFile.vb:11]: Should this be Partial Public?
```suggestion
    Partial Public Class Network
```

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.UploadFile.vb:11]: VB seems to only care if 1 file has the Partial keyword but I agree all should have it. The opposite is also true you can declare a single file Partial and VB doesn't care, but it should be deleted.

I will update all the updated files to use it correctly.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.UploadFile.vb]: Done

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkAvailability.vb (hunk 2 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/NetworkAvailability.vb]: Done

**Review by lonitra — APPROVED** [review]
LGTM 👍

**Pull request merged**